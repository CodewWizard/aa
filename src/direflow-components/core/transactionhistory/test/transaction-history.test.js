import React from "react";
import renderer from "react-test-renderer";
import TransactionHistory from "../transaction-history";

import { render, screen, cleanup, fireEvent } from "@testing-library/react";

afterEach(() => {
  cleanup();
});

const tableRows = [
  {
    transactionType: "Manual Upload",
    documentType: "Data Submission",
    transactionDate: "June 15th, 2023 05:13:18 AM",
    effectiveDate: "June 15th, 2023 05:13:18 AM",
    Uri: "https://devdiep2.azure-api.net/dms/CNGA/M/TX/TOGACNMTX230306/Quote_Prem%20Full%20Test_1.pdf",
  },
  {
    transactionType: "Application",
    documentType: "Submission Data Document",
    transactionDate: "June 15th, 2023 04:20:53 AM",
    effectiveDate: "June 15th, 2023 04:20:53 AM",
    Uri: "https://www.google.com",
  },
];

const tableCols = [
  {
    dataField: "transactionType",
    text: "Transaction Type",
    sort: false,
  },
  {
    dataField: "documentType",
    text: "Document Type",
    sort: false,
  },
  {
    dataField: "transactionDate",
    text: "Transaction Date",
    sort: false,
    hidden: false,
  },
  {
    dataField: "effectiveDate",
    text: "Effective Date",
    sort: false,
    hidden: false,
  },
];

const CommonArgs = {
  title: "Transaction History",
  tableCols,
  tableRows,
  parentClass: "",
  titleClass: "",
  tableClass: "",
};

it("should render given schema correctly", () => {
  const testRenderer = renderer.create(<TransactionHistory {...CommonArgs} />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
  testRenderer.unmount();
});
