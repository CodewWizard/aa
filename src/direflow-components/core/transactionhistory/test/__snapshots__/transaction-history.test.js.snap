// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render given schema correctly 1`] = `
<div
  className="first-table position-relative"
>
  <h3
    className="m-0"
  >
    Transaction History
  </h3>
  <div
    className="w-100"
  >
    <div
      className="singleBox customDataTable"
    >
      <div
        className="react-bootstrap-table"
      >
        <table
          className="table"
        >
          <thead>
            <tr>
              <th
                tabIndex={0}
              >
                Transaction Type
              </th>
              <th
                tabIndex={0}
              >
                Document Type
              </th>
              <th
                tabIndex={0}
              >
                Transaction Date
              </th>
              <th
                tabIndex={0}
              >
                Effective Date
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className={null}
              style={Object {}}
            >
              <td>
                Manual Upload
              </td>
              <td>
                Data Submission
              </td>
              <td>
                June 15th, 2023 05:13:18 AM
              </td>
              <td>
                June 15th, 2023 05:13:18 AM
              </td>
            </tr>
            <tr
              className={null}
              style={Object {}}
            >
              <td>
                Application
              </td>
              <td>
                Submission Data Document
              </td>
              <td>
                June 15th, 2023 04:20:53 AM
              </td>
              <td>
                June 15th, 2023 04:20:53 AM
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
